jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -1735,22 +1735,64 @@ static MethodHandle filterArgument(MethodHandle target, int pos, MethodHandle fi
 	 * @throws IllegalAccessException - if access checking fails
 	 * @throws SecurityException - if the SecurityManager prevents access
 	 */
-	public static MethodHandles.Lookup privateLookupIn(Class<?> targetClass, MethodHandles.Lookup callerLookup) throws IllegalArgumentException, NullPointerException, IllegalAccessException, SecurityException {
-		throw new UnsupportedOperationException("The method has not yet been implemented for now"); //$NON-NLS-1$
+	public static MethodHandles.Lookup privateLookupIn(Class<?> targetClass, MethodHandles.Lookup callerLookup) throws NullPointerException, IllegalArgumentException, IllegalAccessException, SecurityException {
+		if (Objects.isNull(targetClass) || Objects.isNull(callerLookup)) {
+			/*[MSG "K065S", "Both the requested class and the caller lookup must not be null"]*/
+			throw new NullPointerException(com.ibm.oti.util.Msg.getString("K065S")); //$NON-NLS-1$
+		}
+		
+		if (targetClass.isPrimitive() || targetClass.isArray()) {
+			/*[MSG "K065T", "The target class: {0} must not be a primitive type or an array class"]*/
+			throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K065T", targetClass.getCanonicalName())); //$NON-NLS-1$
+		}
+		
+		Module targetClassModule = targetClass.getModule();
+		String targetClassPackageName = targetClass.getPackageName();
+		Module accessClassModule = callerLookup.accessClass.getModule();
+		
+		/* Check whether the named module containing the old lookup can read the module containing the target class.
+		 * Note: an unnamed module can read any module.
+		 */
+		if (!accessClassModule.canRead(targetClassModule)) {
+			/*[MSG "K065U", "The module: {0} containing the old lookup can't read the module: {1}"]*/
+			throw new IllegalAccessException(com.ibm.oti.util.Msg.getString("K065U", accessClassModule.getName(), targetClassModule.getName())); //$NON-NLS-1$
+		}
+		
+		/* Check whether the module has the package (containing the target class) opened to
+		 * the module containing the old lookup.
+		 */
+		if (!targetClassModule.isOpen(targetClassPackageName, accessClassModule)) {
+			/*[MSG "K065V", "The package: {0} containing the target class is not opened to the module: {1}"]*/
+			throw new IllegalAccessException(com.ibm.oti.util.Msg.getString("K065V", targetClassPackageName, accessClassModule.getName())); //$NON-NLS-1$
+		}
+		
+		int callerLookupMode = callerLookup.lookupModes();
+		if (Lookup.MODULE != (Lookup.MODULE & callerLookupMode)) {
+			/*[MSG "K065W", "The access mode: 0x{0} of the caller lookup doesn't have the MODULE mode : 0x{1}"]*/
+			throw new IllegalAccessException(com.ibm.oti.util.Msg.getString("K065W", Integer.toHexString(callerLookupMode), Integer.toHexString(Lookup.MODULE))); //$NON-NLS-1$
+		}
+		
+		SecurityManager secmgr = System.getSecurityManager();
+		if (null != secmgr) {
+			secmgr.checkPermission(com.ibm.oti.util.ReflectPermissions.permissionSuppressAccessChecks);
```

Comment:
There is another occurrence of `new ReflectPermission("suppressAccessChecks")` at method `reflectAs`, pls change it to `ReflectPermissions.permissionSuppressAccessChecks` as well.

Reply:
Updated.